### ISSUE TITLE
Restore AirPlay support for Akamai streams on Apple TV 3rd gen

### ISSUE THREAD

**Issue by defagos** [SRGSSR/pillarbox-apple#133]
**As an Apple TV 3rd gen owner** I would like to be able to use AirPlay when playing any SRG SSR content.

# Acceptance criteria

- All SRG SSR contents can be played with an Apple TV 3rd gen.

# Tasks

- [ ] Check [current behavior](https://github.com/SRGSSR/srgcontentprotection-apple/blob/develop/Sources/SRGContentProtection/SRGAkamaiAssetResourceLoaderDelegate.m) with an Apple TV 3rd gen device.
- [ ] Fix any issues (restoring SRG Content Protection logic if needed).

**Comment by defagos** [comment]
# Tests

I played all demo examples on an Apple TV 3rd gen and  we had in the demo and found 4 issues:

1. The Dig (URL-based content), an MP4 buffering indefinitely on Apple TV 3rd gen. This is not a problem for Pillarbox (or Letterbox) since URN delivery provides an HLS stream which is picked first and plays fine on Apple TV 3rd gen. We cannot also do anything since this is a limitation of Apple TV 3rd gen devices.
2. Couleur 3 live (URL-based content, MP3), also buffering indefinitely on Apple TV 3rd gen. Again this is not a problem since an HLS stream exists and plays just fine. We cannot also do anything since this is a limitation of Apple TV 3rd gen devices.
3. Token-protected sports content: Does not play on Apple TV 3rd gen receivers. Casting from Letterbox demo does not work either.
4. Superfluously token-protected content, but fixing the metadata would fix the issue, also in Letterbox. Also see related [JIRA issue](https://jira.srg.beecollaboration.com/browse/PLAY-4054).

# Conclusion

The SRG Content Protection code we didn't write in the current Pillarbox implementation only was targeting token-protected content, and the behavior is still identical between Pillarbox and Letterbox (mostly does not work).

Fortunately token-protected content is now the exception rather than the rule. DRM content plays just fine with Pillarbox and and Apple TV 3rd gen device, and so does most of the content we have (and if some streams cannot be played, supported streams are available and preferred in general). We should discuss with the backend team to avoid having to fetch a token when not needed, though, as this makes some streams unplayable on old Apple TV devices.

TL;DR
There is no need to import the playlist retrieval logic from SRG Content Protection in Pillarbox.
